app/models/person.rb

```diff
@@ -31,18 +31,28 @@ class Person < ActiveRecord::Base
 
   attr_writer :address, :zip, :remote_fields
 
+  before_create :generate_uuid, unless: :uuid?
   before_save :downcase_email
   after_save :update_nation_builder, :save_location
 
   alias_method :location_association, :location
   delegate :update_location, :district, :state, to: :location
 
+  FIELDS_ALSO_ON_NB = %w[email first_name is_volunteer last_name phone]
+
   def self.create_or_update(person_params)
     if email = person_params.delete(:email)
       find_or_initialize_by(email: email).tap{ |p| p.update(person_params) }
     end
   end
 
+  def self.new_uuid
+    loop do
+      token = SecureRandom.uuid
+      break token unless Person.where(uuid: token).any?
+    end
```

Comment:
this may be overkill and I'm happy to take it out if desired. The likelihood of getting a duplicate uuid is near zero. And, anyway, there's a unique index on the db to prevent record from saving (would result in user data being lost, of course)